### PR TITLE
API: return resource ID when deleting datasource with UID and library element

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What this PR does / why we need it**:
Return resource ID for the following endpoints:
- DELETE `/api/datasources/uid/:uid`
- DELETE `/api/library-elements/:uid`

**Special notes for your reviewer**:
It allows to fill the resource ID field in audit logs (in Enterprise).

### CODE REVIEW

**Review by marefr — APPROVED** [review]
LGTM

File: pkg/services/libraryelements/api.go [hunk 1]
```diff
@@ -35,12 +36,15 @@ func (l *LibraryElementService) createHandler(c *models.ReqContext, cmd CreateLi
...
+	id, err := l.deleteLibraryElement(c.Req.Context(), c.SignedInUser, web.Params(c.Req)[":uid"])
 	if err != nil {
 		return toLibraryElementError(err, "Failed to delete library element")
 	}
 
-	return response.Success("Library element deleted")
+	return response.JSON(200, util.DynMap{
+		"message": "Library element deleted",
+		"id":      id,
+	})
```
Comment (hugohaggmark) [inline pkg/services/libraryelements/api.go:47]: I thought we should avoid using util.DynMap but maybe I'm thinking about something else?

File: pkg/services/libraryelements/database.go [hunk 2]
```diff
@@ -198,8 +199,10 @@ func (l *LibraryElementService) deleteLibraryElement(c context.Context, signedIn
 			return ErrLibraryElementNotFound
 		}
 
+		elementID = element.ID
 		return nil
 	})
+	return elementID, err
```
Comment (hugohaggmark) [inline pkg/services/libraryelements/database.go:205]: Maybe add some tests for this change?
https://github.com/grafana/grafana/blob/main/pkg/services/libraryelements/libraryelements_delete_test.go

**Review by hugohaggmark — COMMENTED** [review]
Looks good, not sure what our process is for changing the format of api results, breaking change label for CHANGELOG?

I think we need to update https://github.com/grafana/grafana/blob/main/docs/sources/http_api/library_element.md#delete-library-element too.

Leaving it up to @grafana/backend-platform to approve

File: pkg/services/libraryelements/api.go (hunk 1 above)
Comment (AgnesToulet) [reply pkg/services/libraryelements/api.go:46]: I didn't know but it makes sense, especially when adding test! I created a structure for that, not sure about the name 🤔

File: pkg/services/libraryelements/database.go (hunk 2 above)
Comment (AgnesToulet) [reply pkg/services/libraryelements/database.go:205]: Done!

**Comment by AgnesToulet** [comment]
I don't think it's a breaking change as this is only adding information but you're right about documentation, I'll update it.

File: pkg/services/libraryelements/api.go (hunk 1 above)
Comment (hugohaggmark) [reply pkg/services/libraryelements/api.go:46]: DeleteLibraryElementResult or LibraryElementDeleteResult?

File: pkg/services/libraryelements/database.go (hunk 2 above)
Comment (hugohaggmark) [reply pkg/services/libraryelements/database.go:205]: Thank you!

**Review by hugohaggmark — APPROVED** [review]
Thanks for adding the tests and updating the docs ⭐

**Review by achatterjee-grafana — APPROVED** [review]

**Pull request merged**